gmso/formats/mol2.py

```diff
@@ -0,0 +1,219 @@
+                        line[1], line[0]
+                    )
+                )
+                charge = None
+            atom = Atom(
+                name=line[1],
+                position=position.to("nm"),
+                charge=charge,
+                element=element,
+                residue_label=line[7],
+                residue_index=int(line[6]),
+            )
+            topology.add_site(atom)
+        else:
+            break
+    return line
+
+
+def _load_top_bonds(f, topology, **kwargs):
+    """Take a mol2 file section with the heading '@<TRIPOS>BOND' and save to the topology.bonds attribute."""
+    while True:
+        line = f.readline()
+        if _if_end_of_rti(line):
+            line = line.split()
+            bond = Bond(
+                connection_members=(
+                    topology.sites[int(line[1]) - 1],
+                    topology.sites[int(line[2]) - 1],
+                )
+            )
+            topology.add_connection(bond)
+        else:
+            break
+    return line
+
+
+def _load_top_box(f, topology, **kwargs):
+    """Take a mol2 file section with the heading '@<TRIPOS>FF_PBC' or '@<TRIPOS>CRYSIN' and save to topology.box."""
+    if topology.box:
+        warnings.warn(
+            "This mol2 file has two boxes to be read in, only reading in one with dimensions {}".format(
+                topology.box
+            )
+        )
+        line = f.readline()
+        return line
+    while True:
+        line = f.readline()
+        if _if_end_of_rti(line):
+            line = line.split()
+            # TODO: write to box information
+            topology.box = Box(
+                lengths=[float(x) for x in line[0:3]] * u.Å,
+                angles=[float(x) for x in line[3:6]] * u.degree,
+            )
+        else:
+            break
+    return line
+
+
+def _parse_record_type_indicator(f, line, topology, site_type):
+    """Take a specific record type indicator (RTI) from a mol2 file format and save to the proper attribute of a gmso topology.
+
+    Supported record type indicators include Atom, Bond, FF_PBC, and CRYSIN.
+    """
+    supported_rti = {
+        "@<TRIPOS>ATOM": _load_top_sites,
+        "@<TRIPOS>BOND": _load_top_bonds,
+        "@<TRIPOS>CRYSIN": _load_top_box,
+        "@<TRIPOS>FF_PBC": _load_top_box,
+    }
+    # read in to atom attribute
+    try:
+        line = supported_rti[line.strip()](f, topology, site_type=site_type)
+    except KeyError:
+        warnings.warn(
+            "The record type indicator {} is not supported. Skipping current section and moving to the next RTI header.".format(
+                line
+            )
+        )
+        line = f.readline()
+    return line
+
+
+def _if_end_of_rti(line):
+    """Check if line in an rti is at the end of the section."""
+    if (
+        "@" not in line
+        and not line == "\n"
+        and line
+        and not line.strip().startswith("#")
```

Comment:
What is the third check for?

Reply:
The third check will check for the end of the file (line will be None it you've reached that point in the readline).

Reply:
If line is `None`, the first condition `'@' in line` will error out.

```py
>>> '@' in None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'NoneType' is not iterable

```
The way to get around this is to include the `NonType` check first, for short circuiting.
```py
>>> a = None
>>> bool(a and '@' in a)
False

```

Reply:
fixed in commit 54e0b7763929045706cdb55984ea77d82e61f5d3